### DISCUSSION TITLE
Create role Programmatically in superset

### DISCUSSION THREAD

**Discussion by milehighdev2** [apache/superset#18378] [Q&A / Help]
Our org has the need to create a power user and read-only role that is different than default roles like gamma and alpha. How do we modify the custom security manager code to create a new type of role, assign permissions and put the user in that role upon the user's first time logging into Superset? We are also using SSO with Keycloak.

**Comment by mayurnewase** [comment]
fab have cli to export and import roles, maybe you can try that?
fab needs to expose this api, so might be better to open this issue there.

**Reply by milehighdev2** [reply]
Can you shed some light on this user was able to do it? Is the roles file he used part of the fab import function you are referring to? 


https://github.com/apache/superset/issues/8864
